packages/nuxt/src/app/composables/payload.ts

```diff
@@ -50,14 +51,14 @@ export function preloadPayload (url: string, opts: LoadPayloadOptions = {}) {
 
 // --- Internal ---
 
-const extension = renderJsonPayloads ? 'json' : 'js'
+const filename = renderJsonPayloads ? '_payload.json' : '_payload.js'
 function _getPayloadURL (url: string, opts: LoadPayloadOptions = {}) {
   const u = new URL(url, 'http://localhost')
   if (u.host !== 'localhost' || hasProtocol(u.pathname, { acceptRelative: true })) {
     throw new Error('Payload URL must not include hostname: ' + url)
   }
-  const hash = opts.hash || (opts.fresh ? Date.now() : '')
-  return joinURL(useRuntimeConfig().app.baseURL, u.pathname, hash ? `_payload.${hash}.${extension}` : `_payload.${extension}`)
+  const hash = opts.hash || (opts.fresh ? Date.now() : (useAppConfig().nuxt as any)?.buildId)
```

Comment:
I haven't tested but this wouldn't affect on domain prerenderin (in runtime) and only append buildId as hash in `prerender` env right?

Reply:
No, this would always add the hash, so that when a pre-rendered payload is re-requested it bypasses browser cache